File: src/main/java/com/haozileung/infra/pager/PageResult.java
package com.haozileung.infra.pager;

import com.google.common.collect.Lists;

import java.io.Serializable;
import java.util.List;

/**
 * Created by haozi on 16-3-9.
 */
public class PageResult<T> implements Serializable {
    private Long total = 0L;
    private List<T> rows = Lists.newArrayList();

    public Long getTotal() {
        return total;
    }

    public void setTotal(Long total) {
        this.total = total;
    }

    public List<T> getRows() {
        return rows;
    }

    public void setRows(List<T> rows) {
        this.rows = rows;
    }
}

File: src/main/kotlin/com/haozileung/infra/page/PageResult.kt
package com.haozileung.infra.page

import com.google.common.collect.Lists
import java.io.Serializable

/**
 * Created by haozi on 16-3-9.
 */
class PageResult<T> (var total: Long? = 0L,
                     var rows: List<T> = Lists.newArrayList<T>()): Serializable {

}
